geomesa-gt/geomesa-gt-partitioning/src/main/scala/org/locationtech/geomesa/gt/partition/postgis/dialect/functions/LogCleaner.scala

```diff
@@ -0,0 +1,61 @@
+/***********************************************************************
+ * Copyright (c) 2013-2022 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.gt.partition.postgis.dialect
+package functions
+
+import org.locationtech.geomesa.gt.partition.postgis.dialect.procedures.{AnalyzePartitions, PartitionMaintenance, RollWriteAheadLog}
+
+/**
+ * Deletes history older than 7 days
+ */
+object LogCleaner extends SqlFunction with CronSchedule {
+
+  override def name(info: TypeInfo): String = "cron_log_cleaner"
+
+  override def jobName(info: TypeInfo): String = s"${info.name}-cron-log-cleaner"
+
+  override protected def createStatements(info: TypeInfo): Seq[String] =
+    Seq(function()) ++ super.createStatements(info)
+
+  override protected def dropStatements(info: TypeInfo): Seq[String] = Seq.empty // function is shared between types
+
+  override protected def schedule(info: TypeInfo): String = "0 0 * * *"
```

Comment:
Errrm.... should the cron schedule be configurable?  Maybe later?

Since this pattern is common, I'd worry that it may run at the same time as other things...

Reply:
yeah, this is a fairly lightweight function... maybe i'll just move it to like 1:30AM? i did make the more process-intensive schedules configurable and semi-randomized when not explicit.

Reply:
If it is lightweight, no worries!